cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -234,13 +236,25 @@ class SimpleArrayMixinCalculators
 
     value_type median_op(small_vector<value_type> & sv) const
     {
+        USE_CALLPROFILER_PROFILE_THIS_SCOPE("SimpleArray::median_op()");
         const size_t n = sv.size();
-        if (n % 2 != 0)
+        T v1;
+        T v2;
+        {
+            USE_CALLPROFILER_PROFILE_THIS_SCOPE("SimpleArray::median_op(): select_kth_1");
```

Comment:
Just add a profiling node in `select_kth()`.

I understand that having a single node in the function scope does not distinguish the two invocations.  It is fine.  You can still tell the number of calls by comparing the call counts between the nodes of `mediam_op()` and `select_kth()`.

Profiling nodes checked in should be concise because they are shown in profiling runs by default.  The scoped nodes are too wordy and should not be checked in.

When working on a specific speedup project, one can manually insert the scoped / detailed profiling nodes without checking them in.